tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/create/CreateApplicationOnBuilderImageTest.java

```diff
@@ -102,6 +113,50 @@ public void testCreateApplicationBasedOnBuilderImage() {
 				+ " of services: " + services.size(), services.size() == 1);
 	}
 	
+    private void validateJBIDE22704() {
+        BuilderImageApplicationWizardHandlingTest.nextToBuildConfigurationWizardPage();
+        
+        applicationName = new LabeledText("Name: ").getText();
+        
+        assertNotNull(applicationName);
+        assertTrue(applicationName.length() > 0);
+        
+        new WaitUntil(new WidgetIsEnabled(new NextButton()));
+
+        /*
+         * switch to the Deployment page
+         */
+        new NextButton().click();
+        new WaitUntil(new WidgetIsEnabled(new BackButton()));
+        
+        int numberofEnvironmentVariables = new DefaultTable().rowCount();
+        assertTrue(numberofEnvironmentVariables > 0);
+        
+        /*
+         * switch to the Routing page
+         */
+        new NextButton().click();
+        new WaitUntil(new WidgetIsEnabled(new BackButton()));
+        
+        int numberOfServicePorts = new DefaultTable().rowCount();
+        assertTrue(numberOfServicePorts > 0);
+        
+        new CancelButton().click();
+        new WaitWhile(new JobIsRunning(), TimePeriod.getCustom(120));
+    }
+    
+    @Test
+    public void validateJBIDE22704FromShellMenu() {
+        new NewOpenShift3ApplicationWizard().openWizardFromShellMenu();
+        validateJBIDE22704();
```

Comment:
My understanding of the code is that the first navigation of the code is performed by openWizardFromShellMenu or openWizardFromCentral so that's why my test succedded